rs/node_rewards/canister/tests/regisry_sync_test.rs

```diff
@@ -1,56 +1,66 @@
 use candid::{Decode, Encode};
 use ic_interfaces_registry::RegistryValue;
-use ic_nns_test_utils::common::{build_node_rewards_test_wasm, NnsInitPayloadsBuilder};
-use ic_nns_test_utils::state_test_helpers::{
-    query, setup_nns_canisters, state_machine_builder_for_nns_tests,
-};
+use ic_nervous_system_integration_tests::pocket_ic_helpers::NnsInstaller;
+use ic_nns_test_utils::common::build_node_rewards_test_wasm;
 use ic_protobuf::registry::subnet::v1::{SubnetRecord, SubnetType};
 use ic_registry_keys::make_subnet_record_key;
 use ic_types::{PrincipalId, SubnetId};
+use pocket_ic::PocketIcBuilder;
 use std::time::Duration;
 
-#[test]
-fn test_registry_value_syncing() {
-    let state_machine = state_machine_builder_for_nns_tests().build();
+#[tokio::test]
+async fn test_registry_value_syncing() {
+    let pocket_ic = PocketIcBuilder::new()
+        .with_nns_subnet()
+        .with_sns_subnet()
+        .build_async()
+        .await;
 
-    let init_payloads = NnsInitPayloadsBuilder::new()
-        .with_initial_invariant_compliant_mutations()
-        .build();
-    setup_nns_canisters(&state_machine, init_payloads);
+    let mut installer = NnsInstaller::default();
+    installer.with_current_nns_canister_versions();
+    installer.install(&pocket_ic).await;
 
     let wasm = build_node_rewards_test_wasm();
 
-    let canister_id = state_machine
-        .install_canister(wasm.bytes(), Encode!().unwrap(), None)
-        .unwrap();
+    let canister_id = pocket_ic.create_canister().await;
+    pocket_ic.add_cycles(canister_id, 100_000_000_000_000).await;
+    pocket_ic
+        .install_canister(canister_id, wasm.bytes(), Encode!().unwrap(), None)
+        .await;
 
     // This is the value from invariant_compliant_mutation
     let test_subnet_id = SubnetId::from(PrincipalId::new_subnet_test_id(999));
-    let response = query(
-        &state_machine,
-        canister_id,
-        "get_registry_value",
-        Encode!(&make_subnet_record_key(test_subnet_id)).unwrap(),
-    )
-    .unwrap();
+    // We don't use agents here because this method is test-only, so we won't
+    // support it beyond this one test.
+    let response = pocket_ic
+        .query_call(
+            canister_id,
+            PrincipalId::new_anonymous().0,
+            "get_registry_value",
```

Comment:
NC

It is a bit weird to me that the Node Rewards canister has a method with this name.

Also, it is sad that we cannot use `Request` to avoid specifying these things.

Reply:
It's a test only method, which is why I don't want to use Request for it.